src/main/java/com/matt/forgehax/mods/StepMod.java

```diff
@@ -29,27 +32,80 @@
           .description("how high you can step")
           .defaultTo(1.2f)
           .min(0f)
+          .success(__ -> {
+            if (isEnabled()) {
+              EntityPlayer player = getLocalPlayer();
+              if (player != null) updateStepHeight(player);
+            }
+          })
+          .build();
+
+  public final Setting<Boolean> unstep =
+      getCommandStub()
+          .builders()
+          .<Boolean>newSettingBuilder()
+          .name("unstep")
+          .description("step down instead of falling")
+          .defaultTo(false)
           .build();
 
   public StepMod() {
     super(Category.PLAYER, "Step", false, "Step up blocks");
   }
 
+  @Override
+  protected void onEnabled() {
+    EntityPlayer player = getLocalPlayer();
+    if (player != null) wasOnGround = player.onGround;
+  }
+
   @Override
   public void onDisabled() {
-    if (getLocalPlayer() != null) {
-      getLocalPlayer().stepHeight = DEFAULT_STEP_HEIGHT;
+    EntityPlayer player = getLocalPlayer();
+    if (player != null) player.stepHeight = DEFAULT_STEP_HEIGHT;
+  }
+
+  void updateStepHeight(EntityPlayer player) {
+    if (player.onGround) {
+      player.stepHeight = stepHeight.get();
+    } else {
+      player.stepHeight = DEFAULT_STEP_HEIGHT;
+    }
+  }
+
+  public boolean wasOnGround = false;
+
+  public void unstep(EntityPlayer player) {
+    if (!MC.isCallingFromMinecraftThread())
+      throw new IllegalStateException("Call this using MC.addScheduledTask");
```

Comment:
🤡

Reply:
It crashes the game if it's called from the wrong thread, this is just a sanity check. Edge cases are important too.

Reply:
Edge cases for what? the impossible? this legit just lags the main thread more lol.

Reply:
if you're concerned about lagging then rewrite the Markers mod. I see no issue here.

Reply:
This method is public too, so it's not impossible. If you want to make it impossible, I can make it private or protected.